elide-core/src/main/java/com/yahoo/elide/core/EntityPermissions.java

```diff
@@ -27,71 +27,74 @@
 import java.util.Arrays;
 import java.util.Collection;
 import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
 import java.util.Map;
-import java.util.concurrent.ConcurrentHashMap;
 
 /**
  * Extract permissions related annotation data for a model.
  */
 @Slf4j
 public class EntityPermissions implements CheckInstantiator {
-    private static final Class[] PERMISSION_ANNOTATIONS = new Class[]{
+    private static final List<Class<? extends Annotation>> PERMISSION_ANNOTATIONS = Arrays.asList(
             ReadPermission.class,
             CreatePermission.class,
             DeletePermission.class,
             SharePermission.class,
             UpdatePermission.class
-    };
+    );
 
     public static final EntityPermissions EMPTY_PERMISSIONS = new EntityPermissions();
 
-    //CHECKSTYLE.OFF: LineLength
-    private final ConcurrentHashMap<Class<? extends Annotation>, ParseTree> classPermissions = new ConcurrentHashMap<>();
-    private final ConcurrentHashMap<String, Map<Class<? extends Annotation>, ParseTree>> fieldPermissions = new ConcurrentHashMap<>();
-    //CHECKSTYLE.ON: LineLength
+    private static final AnnotationBinding EMPTY_BINDING = new AnnotationBinding(null, Collections.emptyMap());
+    private final HashMap<Class<? extends Annotation>, AnnotationBinding> bindings = new HashMap<>();
 
-    private EntityPermissions() {
+    private static class AnnotationBinding {
+        final ParseTree classPermission;
+        final Map<String, ParseTree> fieldPermissions;
+
+        public AnnotationBinding(ParseTree classPermission, Map<String, ParseTree> fieldPermissions) {
+            this.classPermission = classPermission;
+            this.fieldPermissions = fieldPermissions.isEmpty() ? Collections.emptyMap() : fieldPermissions;
+        }
     }
 
-    public EntityPermissions(Class<?> cls, Collection<AccessibleObject> fieldOrMethodList) {
-        initFieldPermissionsMap(fieldOrMethodList);
 
-        for (Class annotationClass : PERMISSION_ANNOTATIONS) {
-            bindClassPermissions(cls, annotationClass);
+    private EntityPermissions() {
```

Comment:
Looks like it's the single constructor for the `EMPTY_PERMISSIONS` (L#47). I suggest we add a comment here.